### PR TITLE
keyfood: fix spider and rename to key_food_us

### PR DESCRIPTION
**Pull request by davidhicks**

### CODE REVIEW

File: locations/spiders/key_food_us.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                # item["brand_wikidata"] = ""
+            elif "halseytradersmarket.keyfood.com" in item["website"]:
+                item["brand"] = "Halsey Traders Market"
+                # item["brand_wikidata"] = ""
+            elif "marketplace.keyfood.com" in item["website"]:
+                item["brand"] = "Marketplace"
+                # item["brand_wikidata"] = ""
+            elif "ernestklein.keyfood.com" in item["website"]:
+                item["brand"] = "Ernest Klein"
+                # item["brand_wikidata"] = ""
```
Comment (Cj-Malone) [inline locations/spiders/key_food_us.py:70]: Strong preference for a map:
```python
    brands = {
        "https://ernestklein.keyfood.com/store": {"brand": "Ernest Klein"},
        "https://fooddynasty.keyfood.com/store": {"brand": "Food Dynasty"},
        "https://foodemporium.keyfood.com/store": {"brand": "The Food Emporium"},
        "https://fooduniverse.keyfood.com/store": {"brand": "Food Universe"},
        "https://galafoods.keyfood.com/store": {"brand": "Gala Foods"},
        "https://galafresh.keyfood.com/store": {"brand": "GalaFresh Farms"},
        "https://halseytradersmarket.keyfood.com/store": {"brand": "Halsey Traders Market"},
        "https://keyfoodmarketplace.keyfood.com/store": {"brand": "Key Food Marketplace"},
        "https://keyfoodstores.keyfood.com/store": {"brand": "Key Food", "brand_wikidata": "Q6398037"},
        "https://marketplace.keyfood.com/store": {"brand": "Marketplace"},
        "https://superfresh.keyfood.com/store": {"brand": "SuperFresh"},
        "https://tropicalsupermarket.keyfood.com/store": {"brand": "Tropical Supermarket"},
        "https://urbanmarketplace.keyfood.com/store": {"brand": "Key Food Urban Marketplace"},
        "https://www.keyfood.com/store": {"brand": "Key Food", "brand_wikidata": "Q6398037"},
    }
```

```suggestion
            if brand := self.brands.get(location["siteUrl"]):
                item.update(brand)
```

File: locations/spiders/key_food_us.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+            apply_category(Categories.SHOP_SUPERMARKET, item)
+
+            item["opening_hours"] = OpeningHours()
+            if location.get("open24Hours"):
+                item["opening_hours"].add_days_range(DAYS, "00:00", "23:59")
+            else:
+                for day_name, day_hours in location["openings"].items():
+                    item["opening_hours"].add_range(
+                        day_name, day_hours.split(" - ", 1)[0], day_hours.split(" - ", 1)[1], "%I:%M %p"
```
Comment (Cj-Malone) [inline locations/spiders/key_food_us.py:80]: I just learnt you can do this:
```suggestion
                        day_name, *day_hours.split(" - ", 1), "%I:%M %p"
```

Comment (Cj-Malone) [reply locations/spiders/key_food_us.py:80]: This way you only have to call `split` ones. While it's not exactly a long operation it is a bit pointless.

Comment (davidhicks) [reply locations/spiders/key_food_us.py]: Thanks, I like that approach. The pattern is repeated very frequently so it'll be a good way to simplify spider code.

**Pull request merged**